carla-carma-integration/carla_carma_bridge/src/carla_carma_bridge/carma_carla_route

```diff
@@ -0,0 +1,146 @@
+from cav_msgs.msg import RouteState
+from cav_msgs.msg import Route
+from cav_msgs.msg import RouteEvent, GuidanceState
+from cav_srvs.srv import GetAvailableRoutes
+from cav_srvs.srv import SetActiveRoute
+
+localized_sub = rospy.Subscriber('/localization/current_pose', PoseStamped)
+state_pub = rospy.Publisher('/guidance/state', GuidanceState)
+selected_route = rospy.get_param('selected_route')
+
+class ErrorStatus:
+    def __init__(self, value, text):
+        self.value = value
+        self.text = text
+
+#Get the routes to be set after localization
+def get_available_routes():
+    service = rospy.ServiceProxy('/get_available_routes', GetAvailableRoutes)     
+    try:
+        resp = service.call(0)
+        available_routes = GetAvailableRoutes()
+        available_routes.availableRoutes = resp.availableRoutes
+
+    except:
+        print("Invalid: /guidance/get_available_routes failed.")
+    else:
+        print("Available Routes acquired")
+        return available_routes.availableRoutes
+
+
+
+def set_route(id):
+    state = GuidanceState()
+    service = rospy.ServiceProxy('/set_active_route', SetActiveRoute)
+    
+    NO_ERROR = ErrorStatus(0, "NO_ERROR")
+    NO_ROUTE = ErrorStatus(1, "NO_ROUTE")
+    ALREADY_FOLLOWING_ROUTE = ErrorStatus(2, "ALREADY_FOLLOWING_ROUTE")
+    ROUTE_FILE_ERROR = ErrorStatus(3, "ROUTE_FILE_ERROR")
+    ROUTING_FAILURE = ErrorStatus(4, "ROUTING_FAILURE")
+    TRANSFORM_ERROR = ErrorStatus(5, "TRANSFORM_ERROR")
+
+    try:
+        #Service Call Request
+        resp = service(0, id)
+        if resp.errorStatus != NO_ERROR.value:
+            errorDescription = ""
+            if resp.errorStatus == NO_ROUTE.value:
+                errorDescription = NO_ROUTE.text
+
+            elif resp.errorStatus == ALREADY_FOLLOWING_ROUTE.value:
+                errorDescription = ALREADY_FOLLOWING_ROUTE.text
+
+            elif resp.errorStatus == ROUTE_FILE_ERROR.value:
+                errorDescription = ROUTE_FILE_ERROR.text
+
+            elif resp.errorStatus == ROUTING_FAILURE.value:
+                errorDescription = ROUTING_FAILURE.text
+
+            elif resp.errorStatus == TRANSFORM_ERROR.value:
+                errorDescription = TRANSFORM_ERROR.text
+        
+            else:
+                errorDescription = resp.errorStatus
+
+        else:
+                #Call succeeded: Setting Route
+                print('call set active route success!')
+                #Change Guidance from READY to ACTIVE
+                #set_guidance_active()
+                   
+
+
+    except:
+            print("Fail")
+    
+    #RouteEvent
+    route_event_sub = rospy.Subscriber('/guidance/route_event', RouteEvent,route_event_callback)
+
+
+def route_event_callback(event):
+    route_event = RouteEvent()
+    route_event.event = event
+    if route_event.event == 6:
+        print("ROUTE_GEN_FAILED")
+        
+def check_selected_route(available_routes):
+    if selected_route in available_routes:
```

Comment:
```
  if A:
    return True
  else
    return False
```
can always be simplified to `return A` since A already has to evaluate to either a truthy value or falsey value for the if statement to work.